linode/manager
# upcoming: [DI-20360] - Updated preferences logic to remove flickering issue

## Description 📝
Changed the current implementation of UserPreference to avoid flickering issue in widget component

## Changes  🔄
List any change relevant to the reviewer.
1. Created a custom hook useAclpPreference in UserPreference.ts file
2. Removed useEffect from CloudPulseWidget component
3. Updated CloudPulseDashboard component to use default time_granularity if preference is not present

## Target release date 🗓️
5th September 2024

## Preview 📷
**Include a screenshot or screen recording of the change**

:bulb: Use `<video src="" />` tag when including recordings in table.

| Before  | After   |
| ------- | ------- |
|![Screenshot 2024-08-29 at 6 58 16 PM](https://github.com/user-attachments/assets/d451815e-71af-4331-a6e9-8d2222e24de1)|![Screenshot 2024-08-29 at 6 58 47 PM](https://github.com/user-attachments/assets/6a3c4ddb-9b16-4318-b1d2-d44e1a8c9949)|
|![Screenshot 2024-08-29 at 6 58 57 PM](https://github.com/user-attachments/assets/e6b9d6c3-4b9a-44ef-bfaf-f9e1aa59c6cb)|![Screenshot 2024-08-29 at 6 58 05 PM](https://github.com/user-attachments/assets/9d387b0a-5d92-4cce-aed2-9a30cc469a06)|


## How to test 🧪
1. Switch to mock user as APIs are not live.
2. Change global filters value & you can check if UI is flickering or not because of component re-renderings

## As an Author I have considered 🤔

*Check all that apply*
